lib/src/native/realm_core.dart

```diff
@@ -3399,25 +3383,30 @@ extension PlatformEx on Platform {
 /// @nodoc
 class SyncErrorDetails {
   final String message;
-  final SyncErrorCategory category;
+  final int _categoryFlag;
   final int code;
-  final String? detailedMessage;
+  final String? path;
   final bool isFatal;
   final bool isClientResetRequested;
   final String? originalFilePath;
   final String? backupFilePath;
   final List<CompensatingWriteInfo>? compensatingWrites;
-  const SyncErrorDetails(
+  SyncErrorDetails(
     this.message,
-    this.category,
+    this._categoryFlag,
     this.code, {
-    this.detailedMessage,
+    this.path,
     this.isFatal = false,
     this.isClientResetRequested = false,
     this.originalFilePath,
     this.backupFilePath,
     this.compensatingWrites,
   });
+
+  // @categoryConstant param should be a value from `realm_error_category` class
+  bool _isFromCategory(int categoryConstant) {
```

Comment:
we usually move the values in realm_core in Dart and use this one. 
This is what we have done with `SyncErrorCodes` enum below. Lets do the same for the categories and make a private enum with the values so we can safelly use it here as parameter type. then we can remove the error prone int value and the comment on this method

Reply:
This also seems unused?

Reply:
Discussed and decided to be removed.